src/special-types-and-traits.md

```diff
@@ -73,7 +51,19 @@ r[lang-types.deref]
 ## `Deref` and `DerefMut`
 
 As well as overloading the unary `*` operator, [`Deref`] and [`DerefMut`] are
-also used in [method resolution] and [deref coercions].
+also used in [deref coercions]; see also [`Receiver`] below.
+
+r[lang-types.receiver]
+## `Receiver`
+
+[`Receiver`] is used in [method resolution]. It indicates that a type may be
+used as a method receiver; that is, the type of a `self` parameter for a
+method. There is a blanket implementation of `Receiver` for all `T: Deref`,
+so it's rare to implement `Receiver` directly: you'd only normally do this
+for smart pointer types which for some reason can't implement `Deref`.
```

Comment:
I'd probably remove this bit about it being rare and instead describe affirmatively why `Receiver` can be implemented but `Deref` cannot in some cases.